lib/inferno/entities/ig.rb

```diff
@@ -0,0 +1,150 @@
+# frozen_string_literal: true
+
+require 'fhir_models'
+require 'pathname'
+require 'rubygems/package'
+require 'zlib'
+
+require_relative '../repositories/igs'
+
+module Inferno
+  module Entities
+    # IG is a wrapper class around the relevant concepts inside an IG.
+    # Not everything within an IG is currently used by Inferno.
+    class IG < Entity
+      ATTRIBUTES = [
+        :id,
+        :profiles,
+        :extensions,
+        :value_sets,
+        :search_params,
+        :examples
+      ].freeze
+
+      include Inferno::Entities::Attributes
+
+      def initialize(params)
+        super(params, ATTRIBUTES)
+
+        @profiles = []
+        @extensions = []
+        @value_sets = []
+        @examples = []
+        @search_params = []
+      end
+
+      def self.from_file(ig_path)
+        raise "#{ig_path} does not exist" unless File.exist?(ig_path)
+
+        # fhir_models by default logs the entire content of non-FHIR files
+        # which could be things like a package.json
+        original_logger = FHIR.logger
+        FHIR.logger = Logger.new('/dev/null')
+
+        if File.directory?(ig_path)
+          from_directory(ig_path)
+        elsif ig_path.end_with? '.tgz'
+          from_tgz(ig_path)
+        else
+          raise "Unable to load #{ig_path} as it does not appear to be a directory or a .tgz file"
+        end
+      ensure
+        FHIR.logger = original_logger if defined? original_logger
+      end
+
+      def self.from_tgz(ig_path)
+        tar = Gem::Package::TarReader.new(
+          Zlib::GzipReader.open(ig_path)
+        )
+
+        ig = IG.new({})
```

Comment:
I think if you make the constructor take `**params` instead of just `*params` you should be able to do a plain `IG.new` without passing an empty hash.